### DISCUSSION TITLE
Execution of function during post trial break

### DISCUSSION THREAD

**Discussion by prajwalgatti** [jspsych/jsPsych#774] [Q&A] [answered]
Hi, all

I want to execute a stimulus generation function (which takes about 800-1200ms to execute) during the inter-trial interval of 2000ms. 
What is a way to achieve this? 

Thanks in advance!

**Comment by jodeleeuw** [comment, marked as answer]
Maybe with the `async` option for the `call-function` plugin? ([docs here](https://www.jspsych.org/plugins/jspsych-call-function/))

```js
var trial = {
	type: 'call-function',
	func: function(done){
		setTimeout(done, 2000); // end the ITI after 2000ms
		generateStimulus();
	},
	async: true
}
```

**Reply by jodeleeuw** [reply]
An advantage of this is that you could also ensure that your stimulus generation function is complete with some slightly fancier logic.
